lib/sass/script/parser.rb

```diff
@@ -479,11 +479,13 @@ def arglist(subexpr, description)
             elsif !keywords.empty?
               raise SyntaxError.new("Positional arguments must come before keyword arguments.")
             end
-
             args << e if e
           end
 
-          return args, keywords, splat unless try_tok(:comma)
+          comma_present = try_tok(:comma)
+          if !comma_present || (@lexer.peek.type == :rparen && comma_present)
+            return args, keywords, splat
+          end
           e = assert_expr(subexpr, description)
```

Comment:
This should be the condition in the loop.

```ruby
while (e = send(subexpr))
  # ...
  return args, keywords, splat unless try_tok(:comma)
end
```

That way you can avoid the initial `send()` call, and you won't have to check to see if the next token is `)`.